### PR TITLE
 [Reproduce tool] Support libFuzzer.

### PR DESCRIPTION
**Pull request by mbarbella-chromium**

### CODE REVIEW

File: src/python/base/modules.py [hunk 1]
```diff
@@ -90,15 +94,13 @@ def fix_module_search_paths():
     sys.path.insert(0, source_directory)
     python_path.insert(0, source_directory)
 
-  # Since paths have now been fixed, import the custom logs module.
-  from metrics import logs
```
Comment (mbarbella-chromium) [inline src/python/base/modules.py]: This was causing some problems for me since it leads to some imports that need the fixes below to work properly. The new behavior seems a bit safer to me, but I'm not completely sure what was different about the paths when running from the tool compared to production. I can investigate more if needed.

**Comment by mbarbella-chromium** [comment]
No rush on this, PTAL when you have time.

File: src/local/butler/reproduce.py [hunk 2]
```diff
@@ -214,33 +218,50 @@ def _prepare_initial_environment(build_directory):
...
+  _copy_root_subdirectory(root_dir, temp_root_dir, 'src')
 
   environment.set_value('CONFIG_DIR_OVERRIDE',
                         os.path.join(temp_root_dir, 'configs', 'test'))
+  # TODO(mbarbella): Don't force the user to set APPENGINE_DIRECTORY explicitly.
+  environment.set_value(
+      'PYTHONPATH',
+      modules.get_pythonpath_separator().join([
+          os.path.join(temp_root_dir, 'src'),
+          environment.get_value('APPENGINE_DIRECTORY')
```
Comment (oliverchang) [inline src/local/butler/reproduce.py]: why is this needed?

File: src/local/butler/reproduce.py [hunk 3]
```diff
@@ -214,33 +218,50 @@ def _prepare_initial_environment(build_directory):
   _copy_root_subdirectory(root_dir, temp_root_dir, 'bot')
```
Comment (oliverchang) [inline src/local/butler/reproduce.py]: would it make sense to share this functionality with the one in butler's run_bot ? Seems better to reduce the duplication here to make things more maintainable.

File: src/python/base/modules.py [hunk 4]
```diff
@@ -90,15 +94,13 @@ def fix_module_search_paths():
...
-
   # Work around google package issues with App Engine SDK.
+  errors = []
```
Comment (oliverchang) [inline src/python/base/modules.py]: Add a comment on why errors is needed.

File: src/python/base/modules.py [hunk 5]
```diff
@@ -47,6 +47,14 @@ def _patch_appengine_modules_for_bots():
     pass
 
 
+def get_pythonpath_separator():
```
Comment (oliverchang) [inline src/python/base/modules.py]: can we just use os.pathsep here?

File: src/local/butler/reproduce.py (hunk 2 above)
Comment (mbarbella-chromium) [reply src/local/butler/reproduce.py]: Found a better way to handle this. There were issues with the import in the libFuzzer launcher's subprocess, but not in the main process. Adding the sdk path to PYTHONPATH instead of updating sys.path was a workaround for it.
